src/Number.php

```diff
@@ -38,6 +38,56 @@ public function __construct($integerPart, $fractionalPart = '')
         $this->fractionalPart = $this->parseFractionalPart((string) $fractionalPart);
     }
 
+    /**
+     * @param $number
+     *
+     * @return self
+     */
+    public static function fromString($number)
+    {
+        $decimalSeparatorPosition = strpos($number, '.');
+        if ($decimalSeparatorPosition === false) {
+            return new self($number, '');
+        }
+
+        return new self(
+            substr($number, 0, $decimalSeparatorPosition),
+            rtrim(substr($number, $decimalSeparatorPosition + 1), '0')
+        );
+    }
+
+    /**
+     * @param float $number
+     *
+     * @return self
+     */
+    public static function fromFloat($number)
+    {
+        if (is_float($number) === false) {
+            throw new \InvalidArgumentException('Floating point value expected');
+        }
+
+        return self::fromString(sprintf('%.14F', $number));
+    }
+
+    /**
+     * @param float|int|string $number
+     *
+     * @return self
+     */
+    public static function fromNumber($number)
+    {
+        if (is_float($number)) {
```

Comment:
I would not use `elseif` here. But rather only a new if statement, or maybe, even a `switch` statement.

```php
switch (gettype($number)) {
  case 'integer':
   ...
}
```

The above saves a one or more function calls.

Reply:
Using gettype for type checking is generally not advised, but using separate ifs is probably cleaner.